src/kili/adapters/kili_api_gateway/asset/operations_mixin.py

```diff
@@ -52,27 +55,52 @@ def list_assets(
             load_asset_json_fields(asset, fields, self.http_client) for asset in assets_gen
         )
 
-        if any("jsonResponse" in field for field in fields):
-            converter = AnnotationsToJsonResponseConverter(
-                graphql_client=self.graphql_client, project_id=filters.project_id
-            )
-            for asset in assets_gen:
-                if "latestLabel.jsonResponse" in fields and asset.get("latestLabel"):
-                    converter.patch_label_json_response(
-                        asset["latestLabel"], asset["latestLabel"]["id"]
-                    )
-                    if added_latest_label_id_field:
-                        asset["latestLabel"].pop("id")
-                if "labels.jsonResponse" in fields:
-                    for label in asset.get("labels", []):
-                        converter.patch_label_json_response(label, label["id"])
-                        if added_label_id_field:
-                            label.pop("id")
+        yield from assets_gen
 
-                yield asset
+    def list_assets_split(
+        self, filters: AssetFilters, fields: ListOrTuple[str], options: QueryOptions, project_info
+    ) -> Generator[Dict, None, None]:
+        """List assets with given options."""
+        options = QueryOptions(
+            options.disable_tqdm, options.first, options.skip, min(options.batch_size, 10)
```

Comment:
Now that we found a better solution, we could maybe put the limit at 100 for LLM projects and at 10 for video projects